### PR TITLE
Add Markdown syntax to render flow and sequence charts.

### PR DESCRIPTION
**Pull request by Lao-liu**
This PR will add supports for chart markdown syntax to generate flow and sequence charts.

flowchat uses  http://flowchart.js.org/

image

sequence uses https://bramp.github.io/js-sequence-diagrams/

image

### CODE REVIEW

**Comment by jonasfranz** [comment]
Please change title and comments/description to english since otherwise this PR will be invalid.

File: templates/base/footer.tmpl [hunk 1]
```diff
@@ -54,6 +54,34 @@
 <!-- Third-party libraries -->
 {{if .RequireHighlightJS}}
 	<script src="{{AppSubUrl}}/vendor/plugins/highlight/highlight.pack.js"></script>
+	<script src="{{AppSubUrl}}/vendor/plugins/flow/raphael.min.js"></script>
+    <script src="{{AppSubUrl}}/vendor/plugins/flow/flowchart.min.js"></script>
```
Comment (lafriks) [inline templates/base/footer.tmpl]: Spacing seems to be incorrect here

File: templates/base/footer.tmpl [hunk 2]
```diff
@@ -54,6 +54,34 @@
...
+    <script src="{{AppSubUrl}}/vendor/plugins/sequence/webfontloader.js"></script>
+    <script src="{{AppSubUrl}}/vendor/plugins/sequence/snap.svg-min.js"></script>
+    <script src="{{AppSubUrl}}/vendor/plugins/sequence/underscore-min.js"></script>
+    <script src="{{AppSubUrl}}/vendor/plugins/sequence/sequence-diagram-min.js"></script>
+
+	<script type="text/javascript">
+        $(document).ready(function () {
+            $('.language-flow').each((i, flow)=>{
```
Comment (lafriks) [inline templates/base/footer.tmpl]: ES6 syntax won't work on MS Internet Explorer

**Review by lafriks — CHANGES_REQUESTED** [review]

**Comment by lunny** [comment]
@Lao-liu I translated your Chinese to English since we are an international community and not everyone know Chinese.

File: templates/base/footer.tmpl (hunk 1 above)
Comment (Lao-liu) [reply templates/base/footer.tmpl]: ok, change use tab replace space.

File: templates/base/footer.tmpl (hunk 2 above)
Comment (Lao-liu) [reply templates/base/footer.tmpl]: don't use ES6 way.

**Comment by lunny** [comment]
Maybe partial fix https://github.com/go-gitea/gitea/issues/3902

**Comment by techknowlogick** [comment]
I worry about adding all of this JS as a dependency. This can be achieved with local template customization for users that need it.

**Comment by lafriks** [comment]
imho this can be easily added in custom template as it is added only in frontend

**Comment by lunny** [comment]
So let's close this one.

**Pull request closed without merging**